### PR TITLE
fix: CI compilation failed on substrait

### LINKED ISSUE #10682: CI compilation failed: cannot find value `default_nullability` in this scope
### Describe the bug

There's a merge conflict with #10640.

### To Reproduce

```sh
error[E0425]: cannot find value `default_nullability` in this scope
    --> datafusion/substrait/src/logical_plan/producer.rs:1414:38
     |
1414 |                         nullability: default_nullability,
     |                                      ^^^^^^^^^^^^^^^^^^^ not found in this scope
```

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jonahgao**
## Which issue does this PR close?

Closes #10682.

## Rationale for this change
A merge conflict exists between #10640 and #10646.

## What changes are included in this PR?
Fix compilation

## Are these changes tested?
Yes
## Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/substrait/src/logical_plan/producer.rs [hunk 1]
```diff
@@ -1411,15 +1411,15 @@ fn to_substrait_type(dt: &DataType, nullable: bool) -> Result<substrait::proto::
                 parameter: Some(parameter::Parameter::DataType(substrait::proto::Type {
                     kind: Some(r#type::Kind::I32(r#type::I32 {
                         type_variation_reference: DEFAULT_TYPE_REF,
-                        nullability: default_nullability,
+                        nullability: r#type::Nullability::Unspecified as i32,
```
Comment (jonahgao) [inline datafusion/substrait/src/logical_plan/producer.rs:1414]: We are not using the nullability of the parameter on the consumer side, so I temporarily change it to `Unspecified`.

**Review by jayzhan211 — APPROVED** [review]
👍

**Review by waynexia — APPROVED** [review]

**Comment by waynexia** [comment]
Thanks @jonahgao

**Comment by jonahgao** [comment]
Thank you for the reviewing @jayzhan211 @waynexia

**Pull request merged**